File: go/observability/observability.go
package observability

import (
	"log"

	"contrib.go.opencensus.io/exporter/stackdriver"
	"go.opencensus.io/plugin/ocgrpc"
	"go.opencensus.io/stats/view"
	"go.opencensus.io/trace"
)

func ConfigureStackdriver(gcpProject string) {
	if err := view.Register(ocgrpc.DefaultClientViews...); err != nil {
		log.Fatalf("Failed to register ocgrpc client views: %v", err)
	}

	sd, err := stackdriver.NewExporter(stackdriver.Options{
		ProjectID: gcpProject,
	})
	if err != nil {
		log.Fatalf("Failed to create Stackdriver exporter: %v", err)
	}
	defer sd.Flush()
	trace.RegisterExporter(sd)
	sd.StartMetricsExporter()
	defer sd.StopMetricsExporter()
	// For demo purposes, always sample
	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
}

File: java/src/main/java/io/grpc/examples/wallet/Observability.java
/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.grpc.examples.wallet;

import io.opencensus.contrib.grpc.metrics.RpcViews;
import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
import io.opencensus.trace.Tracing;
import io.opencensus.trace.config.TraceConfig;
import io.opencensus.trace.samplers.Samplers;

/** Utilities to configure stats and tracing. */
public final class Observability {
  private Observability() {}

  static void registerExporters(String gcpProjectId) {
    try {
      // Register all the gRPC views and enable stats
      RpcViews.registerAllGrpcViews();

      StackdriverStatsExporter.createAndRegister(
          StackdriverStatsConfiguration.builder()
              .setProjectId(gcpProjectId)
              .build());

      // For demo purposes, always sample
      TraceConfig traceConfig = Tracing.getTraceConfig();
      traceConfig.updateActiveTraceParams(
          traceConfig.getActiveTraceParams().toBuilder()
              .setSampler(Samplers.alwaysSample())
              .build());

      StackdriverTraceExporter.createAndRegister(
          StackdriverTraceConfiguration.builder().setProjectId(gcpProjectId).build());
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }
}
